plenum/common/messages/node_messages.py

```diff
@@ -321,15 +321,21 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         if isinstance(self.checkpoint, dict):
             self.checkpoint = Checkpoint(**self.checkpoint)
-        self.batches = [BatchID(*bid) for bid in self.batches]
+        if not self.batches or not isinstance(self.batches[0], dict):
```

Comment:
Adding comment explaining this would be very nice

Reply:
done